### PR TITLE
Static Group Resource Fix: Add default description and add requires-replace for parent_id

### PR DESCRIPTION
**Pull request by rounak-adhikary**
# Description

This  PR fixes the following issues.

### Issue 1 - State inconsistency when creating static group with omitted `description`

With Config
```terraform
data "ome_device" "device_by_ip" {
    filters = {     
         ip_expressions = ["abc", "def"]  
    }
}

resource "ome_static_group" "terraform-group" {
   name       = "terraform-group"
   parent_id  = 1021
   device_ids = data.ome_device.device_by_ip.devices[*].id
}
```

We get Error:
```
ome_static_group.terraform-group: Creating...
╷
│ Error: Provider produced inconsistent result after apply
│
│ When applying changes to ome_static_group.terraform-group, provider "provider[\"registry.terraform.io/dell/ome\"]" produced an unexpected new value:
│ .description: was null, but now cty.StringVal("").
│
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
```

Solution: Make empty string as the default description.

### Issue 2 - State inconsistency while updating group `parent_id`

With Previous Config:
```terraform
resource "ome_static_group" "terraform-group" {
   name       = "terraform-group"
   parent_id  = 1021
   device_ids = ...
   decsription = "something"
}
```
and Current Config:
```terraform
resource "ome_static_group" "terraform-group" {
   name       = "terraform-group"
   parent_id  = 1032
   device_ids = ...
   decsription = "something"
}
```

We get Error:
```
ome_static_group.terraform-group: Modifying... [name=terraform-group]
╷
│ Error: Provider produced inconsistent result after apply
│
│ When applying changes to ome_static_group.terraform-group, provider "provider[\"registry.terraform.io/dell/ome\"]" produced an unexpected new value:
│ .parent_id: was cty.NumberIntVal(1032), but now cty.NumberIntVal(1021).
│
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
```

Solution: This issue occurred because OME API silently failed to update parent id. Thus the fix is to mark the `parent_id` field as requires-replace.

# ISSUE TYPE
Bugfix Pull Request

##### RESOURCE OR DATASOURCE NAME
static group resource

##### OUTPUT

```
root@lglap049:~/terraform-provider-ome# go test terraform-provider-ome/ome -v -run=TestStaticGroup
=== RUN   TestStaticGroup
--- PASS: TestStaticGroup (54.31s)
PASS
ok      terraform-provider-ome/ome      54.322s
root@lglap049:~/terraform-provider-ome# 
root@lglap049:~/terraform-provider-ome# 
root@lglap049:~/terraform-provider-ome# 
root@lglap049:~/terraform-provider-ome# go test terraform-provider-ome/ome -v -run=TestDataSource_ReadGroupDevices
=== RUN   TestDataSource_ReadGroupDevices
--- PASS: TestDataSource_ReadGroupDevices (21.66s)
PASS
ok      terraform-provider-ome/ome      21.677s
root@lglap049:~/terraform-provider-ome#
```

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

### CODE REVIEW

**Comment by terraform-providers-svc** [comment]
Can one of the admins verify this patch?

**Review by Krishnan-Priyanshu — DISMISSED** [review]

File: ome/resource_static_group.go [hunk 1]
```diff
@@ -93,6 +96,9 @@ func (r resourceStaticGroup) Schema(_ context.Context, _ resource.SchemaRequest,
 				MarkdownDescription: "ID of the parent group of the static group.",
 				Description:         "ID of the parent group of the static group.",
 				Required:            true,
+				PlanModifiers: []planmodifier.Int64{
```
Comment (shenda1) [inline ome/resource_static_group.go:99]: Can we document this?

Comment (rounak-adhikary) [reply ome/resource_static_group.go:99]: sure

Comment (rounak-adhikary) [reply ome/resource_static_group.go:101]: done

**Review by shenda1 — APPROVED** [review]

**Review by Krishnan-Priyanshu — APPROVED** [review]

**Pull request merged**